api/envoy/extensions/wasm/v3/wasm.proto

```diff
@@ -92,6 +92,10 @@ message VmConfig {
   // update and do a background fetch to fill the cache, otherwise fetch the code asynchronously and enter
   // warming state.
   bool nack_on_code_cache_miss = 6;
+
+  // The keys of *Envoy's* environment variables exposed to this VM through WASI's environ_get function.
+  // Note that if a key does not exist, then it will be ignored.
+  repeated string env_keys = 7;
```

Comment:
From an API perspective, I wonder if this belongs under some message that encapsulates the "things" we want to allow the VM access to. These might include environment variables, but maybe other proxy attributes. @PiotrSikora do you have an idea what is best here?